src/components/QrcodeCapture.vue

```diff
@@ -9,88 +9,141 @@
+        newWidth = IMG_MAX_SIZE * ratio
+      }
+
+      canvas.value.width = newWidth
+      canvas.value.height = newHeight
+
+      ctx.drawImage(img, 0, 0, canvas.value.width, canvas.value.height)
+
+      revokeObjectURL()
+
+      resolve(canvas)
     }
-  },
-  methods: {
-    async onFileSelect(event) {
-      this.selectedImage = event.target.files[0]
 
-      try {
-        const { BrowserQRCodeReader } = await import('@zxing/browser')
-        this.codeReader = new BrowserQRCodeReader()
-        this.imageBase64 = await this.getImageBase64()
-        this.qrCodeText = await this.tryToDecode()
+    img.onerror = (e) => reject(e)
+
+    img.src = imgUrl
+  })
+
+  promise.finally(revokeObjectURL)
 
-        vibrate.veryShort()
+  return promise
+}
 
-        this.$emit('detect', this.qrCodeText)
-      } catch (err) {
-        vibrate.tripleVeryShort()
+/**
+ * Decode QRCode from canvas.
+ * @returns {<string>}
+ */
+const getQrcode = () => {
+  return codeReader?.decodeFromCanvas(canvas.value!).getText()
+}
 
-        this.$emit('error', err)
+const tryToDecode = () => {
+  return new Promise<string>((resolve, reject) => {
+    // Vue should rerender <canvas> element,
+    // so add a callback to the macrotasks queue
+    setTimeout(() => {
+      try {
+        resolve(getQrcode())
+      } catch (e) {
+        reject(e)
       }
-      // Reset input to trigger change event later if user selects same image (Chrome)
-      this.$refs.fileInput.value = ''
-    },
-
-    /**
-     * Converts image into Base64.
-     * @returns {Promise<string>}
-     */
-    getImageBase64() {
-      return new Promise((resolve, reject) => {
-        const reader = new FileReader()
-
-        reader.onload = (e) => resolve(e.target.result)
-        reader.onerror = (err) => reject(err)
-
-        reader.readAsDataURL(this.selectedImage)
-      })
-    },
-
-    /**
-     * Decode QRCode from Base64 image.
-     * @returns {Promise<string>}
-     */
-    async getQrcode() {
-      const result = await this.codeReader.decodeFromImageElement(this.$refs.imageElement)
-
-      return result.text
-    },
-
-    tryToDecode() {
-      return new Promise((resolve, reject) => {
-        // Vue should rerender <img> element,
-        // so add a callback to the macrotasks queue
-        setTimeout(() => {
-          this.getQrcode()
-            .then((qrCodeText) => resolve(qrCodeText))
-            .catch((err) => reject(err))
-        }, 0)
-      })
+    }, 0)
+  })
+}
+
+const onFileSelect = async (event: Event) => {
+  try {
+    if (!codeReader) {
+      const { BrowserQRCodeReader } = await import('@zxing/browser')
+      codeReader = new BrowserQRCodeReader()
     }
+
+    const input = event.target as HTMLInputElement
+    const file = input.files?.[0]
+
+    if (!file) {
+      throw Error('no file')
+    }
+
+    await drawCanvas(file)
+    qrCodeText = await tryToDecode()
+
+    vibrate.veryShort()
+
+    emit('detect', qrCodeText)
```

Comment:
Intermediate `let qrCodeText` at the top of component probably not required, just remove it